mypy/nodes.py

```diff
@@ -174,6 +196,24 @@ def is_package_init_file(self) -> bool:
         return not (self.path is None) and len(self.path) != 0 \
             and os.path.basename(self.path).startswith('__init__.')
 
+    def serialize(self) -> JsonDict:
+        return {'.class': 'MypyFile',
+                '_name': self._name,
+                '_fullname': self._fullname,
+                'names': self.names.serialize(self._fullname),
+                'is_stub': self.is_stub,
+                }
+
+    @classmethod
+    def deserialize(cls, data: JsonDict) -> 'MypyFile':
```

Comment:
`path` may be needed for some error messages that target other files -- I think that we have a few errors like that.